index.js

```diff
@@ -602,6 +635,81 @@ function packAddress(derivationPath, xpub, hdPassphrase, derivationScheme) {
   )
 }
 
+function getAddressHeader(addressType, networkId) {
+  return Buffer.from([(addressType << 4) | networkId])
+}
+
+function getSpendingPart(pubKey) {
+  return blake2b(pubKey, 28)
+}
+
+function deriveBaseAddress(pubKey, stakePubKey, addressType, networkId, isStakeHash = false) {
+  validateBuffer(pubKey, 32)
+  validateAddressType(addressType, 0, 3)
```

Comment:
what's the purpose of this validation step? Isn't "base" already the type of the address?